src/MGmol.cc

```diff
@@ -1442,6 +1442,33 @@ void MGmol<OrbitalsType>::getAtomicNumbers(std::vector<short>& an)
     ions_->getAtomicNumbers(an);
 }
 
+template <class OrbitalsType>
+void MGmol<OrbitalsType>::updateDMandEnergy(OrbitalsType& orbitals, Ions ions, double& eks)
```

Comment:
You may want to try using a reference to ions here. We do not have a proper copy constructor for the class Ions. @oseikuffuor1 debugging seems to point at that as a possible issue